pkg/cmd/task/start_test.go

```diff
@@ -3962,6 +3962,86 @@ func TestTaskStart_ExecuteCommand(t *testing.T) {
 			wantError:  false,
 			goldenFile: true,
 		},
+		{
+			name: "Dry Run with --use-param-defaults and specified params",
```

Comment:
So this unit test isn't actually testing `--use-param-defaults` because we need to define a param default for a `task-1` param. This test here is just passing in a param value and setting it for the TaskRun. What would be a nice way to test this is having a default value defined for a param (e.g. `myarg`) and another param that also has a default but is set to a value that overrides the default.

Setting params can be done [here](https://github.com/tektoncd/cli/blob/20570c11ae379bf75aaf06bfa604b81e0bbf0563/pkg/cmd/task/start_test.go#L3882). It should be similar to what was done for [`tkn pipeline start` tests](https://github.com/tektoncd/cli/blob/1db36d84bc1ce714bf8956d6450fdc2609a51173/pkg/cmd/pipeline/start_test.go#L698).

Let me know if I can further clarify anything here.